x-pack/qa/sql/no-security/src/test/java/org/elasticsearch/xpack/qa/sql/nosecurity/CliExplainIT.java

```diff
@@ -160,4 +160,22 @@ public void testExplainWithCount() throws IOException {
         assertThat(readLine(), startsWith("}]"));
         assertEquals("", readLine());
     }
+
+    public void testOptimizedPlanWithFoldableLiteralExpressions() throws IOException {
+        index("test", body -> body.field("test_field", "test_value1").field("i", 1));
+        index("test", body -> body.field("test_field", "test_value2").field("i", 2));
+
+        assertThat(command("EXPLAIN (PLAN OPTIMIZED) SELECT POWER(i, (1 + 2)) FROM test"), containsString("plan"));
+        assertThat(readLine(), startsWith("----------"));
+        assertThat(readLine(), startsWith("Project[[POWER(i,3)"));
+        assertThat(readLine(), startsWith("\\_EsRelation[test][i{f}"));
+        assertEquals("", readLine());
+
+        assertThat(command("EXPLAIN (PLAN OPTIMIZED) SELECT POWER(i, (1 - 2)) * ABS(10 + 20) FROM test"),
+            containsString("plan"));
+        assertThat(readLine(), startsWith("----------"));
+        assertThat(readLine(), startsWith("Project[[((POWER(i,(1 - 2))) * ABS((10 + 20)))#"));
```

Comment:
Shouldn't this plan have the constants (1-2 and ABS(10+20)) folded?

Reply:
I have updated the code but force pushed ... (sorry :-( ).
Added this: https://github.com/elastic/elasticsearch/pull/33718/files#diff-0d1503677f70eae9fa10767e44075cecR94 so the inner functions are also folded: https://github.com/elastic/elasticsearch/pull/33718/files#diff-694b8d3dd996b57e56c623dfa0a796fdR170
https://github.com/elastic/elasticsearch/pull/33718/files#diff-694b8d3dd996b57e56c623dfa0a796fdR177